src/swap/useSwapQuote.ts

```diff
@@ -85,10 +170,81 @@ const useSwapQuote = () => {
           : null,
       }
 
-      return updatedExchangeRate
+      const baseApproveTx = createApproveTransaction(updatedField, quote.unvalidatedSwapTransaction)
+      const baseSwapTx = createSwapTransaction(quote.unvalidatedSwapTransaction)
+
+      let maxGasCosts: Array<[TokenBalance, BigNumber]> = []
+      for (const feeCurrency of feeCurrencies) {
+        // TODO: this is Celo specific, make it generic
+        const gasPrice = BigInt((await getGasPrice(feeCurrency.address ?? undefined)).toFixed())
+
+        const approveTx = {
+          ...baseApproveTx,
+          maxFeePerGas: gasPrice,
+          feeCurrency: !feeCurrency.isNative ? (feeCurrency.address as Address) : undefined,
+          // We could estimate this ourselves,
+          // but for now we'll just use the value from the quote
+          gas: BigInt(quote.approveTransaction.gas),
```

Comment:
can we get this another way? i thought that we would be removing this `approveTransaction` in the work Jacob is doing on the backend...

Reply:
Yes I'll check with him.

Reply:
can confirm, going away :)